packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+        body.userIds = values[1];
+        body.spaceNames = spaceNames;
+        body.spaceIds = spaceIds;
+        body.range = range;
+
+        return self.request({
+          method: 'POST',
+          service: 'ediscovery',
+          resource: 'reports',
+          body
+        })
+          .then((res) => res)
+          .catch((reason) => {
+            self.logger.error(`Create Report Error ${reason}`);
+            return Promise.reject(reason);
+          });
+      });
+  },
+
+  /**
+   * Retrievse the user UUIDs associated with each supplied email. If an email with no user
+   * is found an error is thrown. If multiple such emails exist only the first encountered
+   * throws an error.
+   *
+   * @param {Array<String>} emails - A list of emails to resolve
+   * @returns {Array<Promise<UUID>>} The UUIDs associated with the emails
+   */
+  _asUUIDs(emails) {
+    if (!emails) {
+      throw Error('Invalid parameter');
+    }
+
+    if (emails.length <= 0) {
+      return Promise.resolve([]);
+    }
+
+    const promises = [];
+    emails.forEach((userEmail) => {
+      promises.push(
+        // The people list API only supports a single email
+        this.spark.people.list({email: userEmail})
+          .then((res) => {
+            if (res.items.length <= 0) {
+              throw Error(`Unable to find user with email ${userEmail}`);
+            }
+            // User id is base64 encoded and of the format Y2lzY29zcGFyazovL3VzL1BFT1BMRS81ZDU5Yjc5NS02ZmEyLTQ2NTQtOGVjMi03NjlkYjE1YzBkOWU
+            const decodedId = Buffer.from(res.items[0].id, 'base64').toString();
+            // Decode and strip out the uuid ciscospark://us/PEOPLE/5d59b795-6fa2-4654-8ec2-769db15c0d9e
+            const uuid = decodedId.substring(decodedId.lastIndexOf('/') + 1, decodedId.length);
+            return uuid;
+          })
+          .catch((reason) => {
+            this.logger.error(`Error converting UUIDs: ${reason}`);
+            return Promise.reject(reason);
+          })
+      );
+    });
+    return Promise.all(promises);
+  },
+
+  /**
+   * Encrypts any potentially sensitive information relating to the report request
+   * The encrypted info is then sent to the eDiscovery service and only decrypted when retrieved again by the client
+   *
+   * @param {String} name - A label to identify the report
+   * @param {String} description - A textual summary of the reports purpose
+   * @param {Array<String>} keywords - A list of search terms relevant to the report
+   * @returns {Promise<encryptedRequestInfo>} Object containing encrypted versions of the original parameters
+   */
+  _encryptRequestParams(name, description, keywords) {
+    const encryptedRequestInfo = {
+      encryptedName: '',
+      encryptedDescription: '',
+      encryptedKeywords: [],
+      encryptionkeyUrl: ''
+    };
+
+    if (!name && !description && (!keywords || keywords.length <= 0)) {
+      return Promise.resolve(encryptedRequestInfo);
+    }
+
```

Comment:
A key is created and bound to a KRO in order to ensure it does not expire while the report is still valid.